### ISSUE TITLE
class RealmModel is not part of the schema for this Realm.

### ISSUE THREAD

**Issue by Gkundu1** [realm/realm-java#4616]
Recently change the realm plugin for android from 3.1.0. to 3.1.4

here how the project/build.gradle look like

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    repositories {
        jcenter()

    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.3.1'
        classpath "io.realm:realm-gradle-plugin:3.1.4"

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
        classpath 'com.google.gms:google-services:3.0.0'
    }
}

allprojects {
    repositories {
        jcenter()

    }

}

task clean(type: Delete) {
    delete rootProject.buildDir
}
```

here is the dependencies look like for app/build.gradle

```
buildscript {
    repositories {
        maven { url 'https://maven.fabric.io/public' }
    }

    dependencies {
        // These docs use an open ended version so that our plugin
        // can be updated quickly in response to Android tooling updates

        // We recommend changing it to the latest version from our changelog:
        // https://docs.fabric.io/android/changelog.html#fabric-gradle-plugin
        classpath 'io.fabric.tools:gradle:1.+'
    }
}
apply plugin: 'com.android.application'
apply plugin: 'io.fabric'
apply plugin: 'realm-android'


repositories {
    maven { url 'https://maven.fabric.io/public' }
}

android {
    compileSdkVersion 25
    buildToolsVersion "25.0.2"
    defaultConfig {
        applicationId "com.ai.telepathy"
        minSdkVersion 15
        targetSdkVersion 16
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
        multiDexEnabled true
    }
    dexOptions {
        preDexLibraries = false
    }
    buildTypes {
        release {
            minifyEnabled false
            zipAlignEnabled true
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }

        debug {
            debuggable true
            jniDebuggable true
            zipAlignEnabled true
        }
    }

    android {
        lintOptions {
            disable 'TypographyFractions', 'TypographyQuotes'
            checkReleaseBuilds false
            // Or, if you prefer, you can continue to check for errors in release builds,
            // but continue the build even when errors are found:
            abortOnError false
        }
    }

    android {
        packagingOptions {
            exclude 'META-INF/DEPENDENCIES.txt'
            exclude 'META-INF/LICENSE.txt'
            exclude 'META-INF/NOTICE.txt'
            exclude 'META-INF/NOTICE'
            exclude 'META-INF/LICENSE'
            exclude 'META-INF/DEPENDENCIES'
            exclude 'META-INF/notice.txt'
            exclude 'META-INF/license.txt'
            exclude 'META-INF/dependencies.txt'
            exclude 'META-INF/LGPL2.1'
        }
    }
}

repositories {
    // You can also use jcenter if you prefer
    mavenCentral()
}

dependencies {
    compile fileTree(dir: 'libs', include: ['*.jar'])
    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
        exclude group: 'com.android.support', module: 'support-annotations'
    })
    compile('com.crashlytics.sdk.android:crashlytics:2.6.7@aar') {
        transitive = true;
    }
    // Android SDK
    //noinspection GradleCompatible
    //noinspection GradleCompatible
    compile 'com.amazonaws:aws-android-sdk-s3:2.2.14'
    compile 'com.amazonaws:aws-android-sdk-s3:2.2.14'
    //noinspection GradleCompatible
    compile 'com.android.support:appcompat-v7:25.3.1'
    compile 'com.android.support:design:25.3.1'
    compile 'com.android.support:recyclerview-v7:25.3.1'
    compile 'com.android.support:cardview-v7:25.3.1'
    compile 'com.android.support.constraint:constraint-layout:1.0.2'
    compile 'com.android.support:multidex:1.0.1'
    compile 'com.google.code.gson:gson:2.8.0'
    compile 'com.squareup.retrofit2:retrofit:2.2.0'
    compile 'com.squareup.retrofit2:converter-gson:2.1.0'
    compile 'com.rom4ek:arcnavigationview:1.0.3'
    compile 'de.hdodenhof:circleimageview:2.1.0'
    compile 'com.github.bumptech.glide:glide:3.7.0'
    compile 'com.android.support:support-v4:25.3.1'
    compile 'com.facebook.android:facebook-android-sdk:4.21.1'
    compile 'com.google.firebase:firebase-messaging:10.2.1'
    compile 'com.google.android.gms:play-services-auth:10.2.1'
    compile 'com.google.android.gms:play-services-plus:10.2.1'
    compile 'co.infinum:materialdatetimepicker-support:3.1.3'
    compile 'com.jaredrummler:android-device-names:1.1.2'
    compile 'com.amazonaws:aws-android-sdk-core:2.2.14'
    compile 'com.amazonaws:aws-android-sdk-cognito:2.2.14'
    compile 'com.amazonaws:aws-android-sdk-ddb:2.2.14'
    compile 'com.crystal:crystalrangeseekbar:1.1.1'
    testCompile 'junit:junit:4.12'
}

apply plugin: 'com.google.gms.google-services'
```

Before upgrading to 3.1.4 ,the saving of data and also retrieve data from database.
But after upgrading to 3.1.4 now every time when try to save the data its only shows
SelectProductModel is not part of the schema for this Realm 

here is the code snippet for SelectProductModel class

```
public class SelectProductModel extends RealmObject
{
    @Required
    private String productId;
    @Required
    private String productName;
    @Required
    private String productPrice;
    @Required
    private String productDesc;
    @Required
    private String productImgUrl;

    public String getProductId() {
        return productId;
    }

    public void setProductId(String productId) {
        this.productId = productId;
    }

    public String getProductName() {
        return productName;
    }

    public void setProductName(String productName) {
        this.productName = productName;
    }

    public String getProductPrice() {
        return productPrice;
    }

    public void setProductPrice(String productPrice) {
        this.productPrice = productPrice;
    }

    public String getProductDesc() {
        return productDesc;
    }

    public void setProductDesc(String productDesc) {
        this.productDesc = productDesc;
    }

    public String getProductImgUrl() {
        return productImgUrl;
    }

    public void setProductImgUrl(String productImgUrl) {
        this.productImgUrl = productImgUrl;
    }
}
```

here is code snippet where saving of data takes place

```
final Realm realm=Realm.getDefaultInstance();
            //realm.beginTransaction();
            realm.executeTransactionAsync(new Realm.Transaction() {
                @Override
                public void execute(Realm realm) {
                    SelectProductModel selectProductModel=realm.createObject(SelectProductModel.class);
                    selectProductModel.setProductId(id);
                    selectProductModel.setProductName(name);
                    selectProductModel.setProductPrice(price);
                    selectProductModel.setProductDesc(description);
                    selectProductModel.setProductImgUrl(imageUrl);
                    realm.commitTransaction();
                }
            }, new Realm.Transaction.OnSuccess() {
                @Override
                public void onSuccess() {
                    callback.onSuccess(id);
                }
            }, new Realm.Transaction.OnError() {
                @Override
                public void onError(Throwable error) {
                    LoggerUtils.error(OSProductRepository.class.getSimpleName(),"realm database error "+error.getMessage());
                    callback.onError("Error while product insert"+error.getMessage());
                }
        });
```

in the above code if uncomment the **realm.beginTransaction()**;  line not show the error but transaction goes for infinite time.
here in the configuration for realm in android application class

```
Realm.init(getApplicationContext());
        RealmConfiguration config = new RealmConfiguration.Builder()
                .name("Telepathy.realm")
                .deleteRealmIfMigrationNeeded()
                .modules(new SelectProductModule())
                .modules(new OSCartModule())
                .schemaVersion(1)
                .build();
        Realm.setDefaultConfiguration(config);
```

Please provide some guidance and solution to overcome this problem.

**Comment by zaki50** [comment]
`RealmConfiguration.Builder.modules()` clears modules list at first.
You need to pass multiple modules in one call.

```
                .modules(new SelectProductModule())
                .modules(new OSCartModule())
```
should be
```
                .modules(new SelectProductModule(), new OSCartModule())
```

**Comment by bmeike** [comment]
Hey @Gkundu1,
  @zaki50 's answer should do the trick.  I'm going to close this ticket.
  Please feel free to re-open it if you need more info or to open a new ticket if you need other help!
-blake

**Comment by 1110101** [comment]
Same issue here on release builds (with proguard enabled). No issues with debug build.
Updated from 3.1.4 to 3.2.0/3.2.1.

`java.lang.RuntimeException: Unable to start activity ComponentInfo{com.xxx/com.xxx.activities.MainActivity}: java.lang.IllegalArgumentException: Invalid query: table 'SC1' not found in this schema.`
`[..]`
`Caused by: java.lang.IllegalArgumentException: Invalid query: table 'SC1' not found in this schema.`


        

            Realm.init(getApplicationContext());

		RealmConfiguration config;

		config = new RealmConfiguration.Builder()
				.schemaVersion(AppConstants.REALM_SCHEMA)
				.modules(new DefaultModule())
				.migration(new NFRealmMigration())
				.build();

		Realm.setDefaultConfiguration(config);
`

Tried to make a proguard rule without success, but im not sure if it was done correctly.

**Comment by Zhuinden** [comment]
Do you have a class named SC1?

**Comment by 1110101** [comment]
Yes of course

    @RealmClass
    public class SC1 extends RealmObject {

	@Required
	@PrimaryKey
	private String rowId;

    [..]

**Comment by 1110101** [comment]
Resolved with 3.3.0
#4673
